perf-system/generator/generate_packages.py

```diff
@@ -0,0 +1,71 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the Apache 2.0 License.
+
+import argparse
+import generator
+
+
+def main():
+    """
+    The function to receive the arguments
+    from the command line
+    """
+    arg_host = "https://127.0.0.1:8000"
+    arg_path = "/app/log/private"  # default path to request
+    arg_type = "HTTP/1.1"  # default type
+    arg_verb = "POST"  # default verb
+    arg_iterations = 16
+    arg_parquet_filename = "requests.parquet"
+    arg_data = '{"id": 1, "msg": "Send message with id 1"}'
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "-hs",
+        "--host",
+        help="The host to submit the request. Default `https://127.0.0.1:8000`",
+        type=str,
+    )

```

Comment:
Repeating the default value risks it getting out-of-sync (need to update it in 2 places, keep the type consistent, keep the formatting consistent each place you define a default). You can instead pass a default directly to this `add_argument` call, and then use `formatter_class=argparse.ArgumentDefaultsHelpFormatter` so that it is displayed in the help output:

```suggestion
    parser = argparse.ArgumentParser(description="Generator for perf workloads", formatter_class=argparse.ArgumentDefaultsHelpFormatter)
    parser.add_argument(
        "-hs",
        "--host",
        help="The host to submit the request.",
        default="https://127.0.0.1:8000",
        type=str,
    )
```

Then when you want to use the parsed-argument-or-the-default, you can just use `args.host` rather than `args.host or default_host`

Reply:
Great suggestion @eddyashton. I missed the default argument on argparse lib.